File: src/Tests/Behavioral/ForVariants.cs
// package main -- go2cs converted at 2018 August 09 13:23:02 UTC
// Original source: D:\Projects\go2cs\src\Tests\Behavioral\ForVariants.go
using fmt = go.fmt_package;
using static go.builtin;

namespace go
{
    public static partial class main_package
    {
        private static void Main()
        {

            var i = 0;

            while(i < 10)
            {
                // Inner comment
                f(i); // Call function
                // Increment i
                i++; // Post i comment
            } // Post for comment

            fmt.Println();
            fmt.Println("i =", i);
            fmt.Println();

            i = 0;

            while(i < 10)
            {
                f(i);

                {
                    var j = 0;

                    while(j < 3)
                    {
                        f(i + j);
                        j++;
                    }
                }
                fmt.Println();
                i++;
            }

            fmt.Println("i =", i);
            fmt.Println();

            {
                var i = 0;

                while(i < 5)
                {
                    // a
                    f(i); // b
                    i++;
                }
            } //c

            fmt.Println();
            fmt.Println("i =", i);
            fmt.Println();

            while(true)
            {
                i++;
                f(i);

                if (i > 12)
                {
                    break;
                }
            }

            fmt.Println();
            fmt.Println("i =", i);
        }

        private static void f(@int y)
        {
            fmt.Print(y);
        }
    }
}

File: src/Tests/Behavioral/ForVariants.go
package main

import (
	"fmt"
)

func main() {

	i := 0
	
	for ; i < 10; {
	    // Inner comment
		f(i) // Call function
		// Increment i
		i++ // Post i comment
	} // Post for comment
	
	fmt.Println()
	fmt.Println("i =", i)
	fmt.Println()
	
	i = 0
	
	for ; i < 10; i++ {
		f(i)
		
		for j := 0; j < 3; j++ {
			f(i + j)
		}
	    fmt.Println()
	}
	
	fmt.Println("i =", i)
	fmt.Println()

	for i := 0; i < 5; i++ {
		// a
		f(i) // b
	} //c
	
	fmt.Println()
	fmt.Println("i =", i)
	fmt.Println()
	
	for {
		i++
		f(i)
		
		if i > 12 {
		    break
		}
	}
	
	fmt.Println()
	fmt.Println("i =", i)
}

func f(y int) {
	fmt.Print(y)
}
